docs/how-to/package-with-hatch.md

```diff
@@ -0,0 +1,385 @@
+In the second line above, before the `=` character, `calculator` is the name of the command.
+The string after the `=` character contains the name of the import (`textual_calculator.entry_points`), followed by a colon (`:`), and then the name of the function (also called `calculator`).
+
+Specifying an entry point like this is equivalent to doing the following from the Python REPL:
+
+```
+>>> import textual_calculator.entry_points
+>>> textual_calculator.entry_points.calculator()
+```
+
+To add the `calculator` command once you have edited `pyproject.toml`, run the following from the command line:
+
+```bash
+pip install -e .
+```
+
+!!! info
+
+    You will have no doubt used `pip` before, but perhaps not with `-e .`.
+    The addition of `-e` installs the project in *editable* mode which means pip won't copy the `.py` files code anywhere, the dot (`.`) indicates were installing the project in the current directory. 
+
+Now you can launch the calculator from the command line as follows:
+
+```bash
+calculator
+```
+
+## Building 
+
+Building produces archive files that contain your code.
+When you install a package via pip or other tool, it will download one of these archives.
+
+To build your project with Hatch, change to the directory containing your `pyproject.toml` and run the `hatch build` subcommand:
+
+```
+cd textual-calculator
+hatch build
+```
+
+After a moment, you should find that Hatch has created a `dist` (distribution) folder, which contains the project archive files.
+You don't typically need to use these files directly, but feel free to have a look at the directory contents.
+
+!!! note "Packaging TCSS and other files"
+
+    Hatch will typically include all the files needed by your project, i.e. the `.py` files.
+    It will also include any Textual CSS (`.tcss`) files in the project directory.
+    Not all build tools will include files other than `.py`; if you are using another build tool, you may have to consult the documentation for how to add the Textual CSS files.
+
+
+## Publishing
+
+After your project has been successfully built you are ready to publish it to PyPI.
+
+If you don't have a PyPI account, you can [create one now](https://pypi.org/account/register/).
+Be sure to follow the instructions to validate your email and set up 2FA (Two Factor Authentication).
+
+Once you have an account, login to PyPI and go to the Account Settings tab.
+Scroll down and click the "Add API token" button.
+In the "Create API Token" form, create a token with name "Uploads" and select the "Entire project" scope, then click the "Create token" button.
+
+Copy this API token (long string of random looking characters) somewhere safe.
+This API token is how PyPI authenticates uploads are for your account, so you should never share your API token or upload it to the internet.
+
+Run the following command (replacing `<YOUR API TOKEN>` with the text generated in the previous step):
+
+```bash
+hatch publish -u __token__ -a <YOUR API TOKEN>
```

Comment:
Aside: This must be considered bad security practice vs env vars because your API token is going to be in plain text in your shell history and will be more likely to leak.

Reply:
How do you set the env var without it going in to the history then?

Would it be better to omit the `-a` and have Hatch ask us for the key?

Reply:
Either via your shell startup file as an env var, having Hatch ask for the key, or removing it from your history after running. Not necessarily saying you should change this btw, it was just a note.